backend/services/eventService.js

```
@@ -0,0 +1,273 @@
+        event.registeredUsers.push(userId);
+        event.seatsRemaining--;
+        await event.save();
+
+        if (!user.eventsJoined.includes(event._id)) {
+            user.eventsJoined.push(event._id);
+            await user.save();
+        }
+
+        return event;
+    } catch (error) {
+        throw error;
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)